src/main/java/pl/wkos/recipebook/service/RecipeService.java

```diff
@@ -0,0 +1,127 @@
+package pl.wkos.recipebook.service;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.tomcat.util.http.fileupload.IOUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import pl.wkos.recipebook.model.Category;
+import pl.wkos.recipebook.model.Recipe;
+import pl.wkos.recipebook.repository.CategoryRepository;
+import pl.wkos.recipebook.repository.RecipeRepository;
+
+import java.io.*;
+import java.util.List;
+
+@Getter
+@Setter
+@Service
+public class RecipeService {
+    RecipeRepository recipeRepository;
+    CategoryRepository categoryRepository;
+
+    @Autowired
+    public RecipeService(RecipeRepository recipeRepository, CategoryRepository categoryRepository) {
+        this.recipeRepository = recipeRepository;
+        this.categoryRepository = categoryRepository;
+    }
+
+    public void addLike(Long id){
+        Recipe recipe = recipeRepository.getRecipeById(id);
+        recipe.setNumberOfLikes(recipe.getNumberOfLikes()+1);
+        recipeRepository.save(recipe);
+    }
+
+    public void addRecipe(MultipartFile file, Recipe recipe){
+        try {
+            File oFile = new File("src/main/resources/static/images/recipes/" + file.getOriginalFilename());
+            OutputStream os = new FileOutputStream(oFile);
+            InputStream inputStream = file.getInputStream();
+            IOUtils.copy(inputStream, os);
+            recipe.setPictureFileName(file.getOriginalFilename());
+            os.close();
+            inputStream.close();
```

Comment:
To jest okej - w rzeczywistym systemie możesz zapisywać binaria (obrazki itp) po stronie bazy danych.